aml-development/frontend
# feat(storefront): allow user to customize storefront layout via profile

AMLOS-743

**Description**     
As a user, I would like to customize the view of my storefront in my profile.  (Reorder/hide/display right side components - Frequently Visited, Updates, My Listings)

Possible starting point: https://github.com/aml-development/frontend/compare/customize_storefront

**Acceptance Criteria**   
The storefront view is customizable through the user profile.

This PR is for the backend functionality necessary for this feature

**How to Test**
Pull `customize_storefront` from `frontend`
Pull `customize_storefront` from `ozp-backend`.  Make sure to `make dev run` on the backend

1. Log in as any user and edit your profile
2. In the _Preferences_ section, look for the "Storefront Layout" option
3. Drag-and-drop items to re-order them.  Click the eye button to show/hide the section.  When finished, save the profile.
4. Return to storefront.  Ensure that the storefront layout updates appropriately.
5. Go back to the edit profile page and click the "Restore Defaults" button next to the "Storefront Layout" header.  Save the profile.
6. Return to storefront.  Ensure that the storefront is back to the default configuration.
